holopy/core/io/io.py

```diff
@@ -64,35 +64,25 @@ def load(inf, spacing = None, optics = None, channel=None):
     """
     if isinstance(optics, (str, IOBase)):
         optics = serialize.load(optics)
-        # In the past We allowed optics yamls to be written without an !Optics
-        # tag, so for that backwards compatability, we attempt to turn an
-        # anonymous dict into an Optics
-        if isinstance(optics, dict):
-            optics = Optics(**optics)
 
     loaded_yaml = False
+    # attempt to load a holopy yaml file
     try:
-         loaded = serialize.load(inf)
-         loaded_yaml = True
+        loaded = serialize.load(inf)
+        if spacing is not None or wavelen is not None or index is not None or polarization is not None or optics is not None:
+            warn("WARNING: If you are trying to overwrite hologram parameters, you must do so explicitly. Extra arguments are being ignored.") 
+        loaded_yaml = True
```

Comment:
I don't understand the logic here. It looks like you are saying that you are ignoring the arguments the user supplies, but in the code that follows it looks like you do you them.

Also, what do you mean by "explicitly"? Something other than passing them directly as arguments to the function?